File: csharp/DiagramPrinter/DiagramPagesReport.cs
namespace DiagramPrinter;

public class DiagramPagesReport
{
    private List<DiagramReportPage> pages = new List<DiagramReportPage>();

    public void AddPage(DiagramReportPage page)
    {
        this.pages.Add(page);
    }

    public bool IsComplete()
    {
        return pages.Count != 0;
    }
}

File: kotlin/src/main/kotlin/sammancoaching/DiagramPagesReport.kt
package sammancoaching

class DiagramPagesReport {
    private val pages: MutableList<DiagramReportPage> = mutableListOf()

    fun addPage(page: DiagramReportPage) {
        pages.add(page)
    }

    fun isComplete(): Boolean {
        return pages.isNotEmpty()
    }
}